lib/storage/partition.go

```diff
@@ -434,7 +435,21 @@ func (pt *partition) AddRows(rows []rawRow) {
 			}
 		}
 	}
+	if pt.s.enableDailyPartitioning {
```

Comment:
would it make sense to check if []rawRow contains timestamps from different dates and only then sort them? 
I expect most of batches will belong to one day, so if we can avoid sorting by just checking their timestamps first - that should speed up the happy path.

Another point is that it should be easier to not sort, but immediately partition entries. Partition for today and yesterday could contain unsorted items. That should speed up things too.